packages/connect-web/src/connect-transport.ts

```diff
@@ -200,6 +200,14 @@ export function createConnectTransport(
           const [demuxedHeader, demuxedTrailer] = trailerDemux(
             response.headers
           );
+
+          // Ignore unknown fields by default if we need to parse JSON format.
+          // Note this is done automatically via the createClientMethodSerializers call above,
+          // but this is a special path that doesn't use that function so we need to set it
+          // explicitly here.
+          if (options.jsonOptions) {
+            options.jsonOptions.ignoreUnknownFields ??= true;
+          }
```

Comment:
This code block does not run if no `jsonOptions` are given, and unknown fields will not be ignored.